src/files/glob-source.js

```diff
@@ -7,131 +7,71 @@ const Path = require('path')
 const errCode = require('err-code')
 
 /**
- * Create an async iterator that yields paths that match requested file paths.
+ * Create an async iterator that yields paths that match requested glob pattern
  *
- * @param {Iterable<string> | AsyncIterable<string> | string} paths - File system path(s) to glob from
+ * @param {string} pattern - Glob pattern to match
  * @param {Object} [options] - Optional options
- * @param {boolean} [options.recursive] - Recursively glob all paths in directories
+ * @param {string} [options.cwd] - The directory to start matching the pattern in
  * @param {boolean} [options.hidden] - Include .dot files in matched paths
- * @param {Array<string>} [options.ignore] - Glob paths to ignore
  * @param {boolean} [options.followSymlinks] - follow symlinks
  * @param {boolean} [options.preserveMode] - preserve mode
  * @param {boolean} [options.preserveMtime] - preserve mtime
  * @param {number} [options.mode] - mode to use - if preserveMode is true this will be ignored
  * @param {import('ipfs-unixfs').MtimeLike} [options.mtime] - mtime to use - if preserveMtime is true this will be ignored
  * @yields {Object} File objects in the form `{ path: String, content: AsyncIterator<Buffer> }`
  */
-module.exports = async function * globSource (paths, options) {
+module.exports = async function * globSource (pattern, options) {
   options = options || {}
 
-  if (typeof paths === 'string') {
-    paths = [paths]
+  if (typeof pattern !== 'string') {
+    throw errCode(
+      new Error('Pattern must be a string'),
+      'ERR_INVALID_PATH',
+      { pattern }
+    )
   }
 
-  const globSourceOptions = {
-    recursive: options.recursive,
-    glob: {
-      dot: Boolean(options.hidden),
-      ignore: Array.isArray(options.ignore) ? options.ignore : [],
-      follow: options.followSymlinks != null ? options.followSymlinks : true
+  let cwd = options.cwd
+
+  if (!cwd) {
+    if (Path.isAbsolute(pattern)) {
+      cwd = Path.dirname(pattern)
+      pattern = pattern.replace(cwd + Path.sep, '')
+    } else {
+      cwd = process.cwd()
     }
   }
 
-  // Check the input paths comply with options.recursive and convert to glob sources
-  for await (const path of paths) {
-    if (typeof path !== 'string') {
-      throw errCode(
-        new Error('Path must be a string'),
-        'ERR_INVALID_PATH',
-        { path }
-      )
-    }
+  if (pattern.startsWith('.' + Path.sep)) {
+    pattern = pattern.replace('.' + Path.sep, '')
```

Comment:
What does this bit guard against?

Reply:
With hindsight I don't think it's necessary, I've removed it.